game-core/src/main/java/games/strategy/engine/random/RandomStatsDetails.java

```diff
@@ -66,10 +66,10 @@ private static DiceStatistic getDiceStatistic(final IntegerMap<Integer> stats, f
       }
       // calculate variance
       double sumOfSquaredMeanDeviations = 0;
+      final double ratio = (double) localTotal / diceSides;
       // TODO: does this need to be updated to take data.getDiceSides() ?
       for (int i = 1; i <= diceSides; i++) {
-        sumOfSquaredMeanDeviations += (double) (stats.getInt(i) - (localTotal / diceSides))
-            * (stats.getInt(i) - (localTotal / diceSides));
+        sumOfSquaredMeanDeviations += Math.pow(stats.getInt(i) - ratio, 2);
```

Comment:
I randomly generated dice rolls on a d6 for various counts and calculated the variance:

Rolls | Variance (old) | Variance (new) | % Change
:-- | :-- | :-- | :--
10 | 0.666667 | 0.370370 | -44.44
100 | 1.17172 | 1.14478 | -2.30
1,000 | 0.444444 | 0.441775 | -0.60
10,000 | 0.999700 | 0.999433 | -0.03
100,000 | 0.467985 | 0.467958 | -0.01

Obviously, the difference tends towards zero as the number of rolls increases.

AFAICT, this value is only used to display statistics in the **Random Stats** dialog.  So, it may not be important that the stats have changed for rolls < 1,000.  As more and more turns are played in a game, the results are going to eventually converge.